### PR TITLE
Add a .generated_files

### PR DESCRIPTION
**Pull request by nmuesch**
### What does this PR do?

This PR adds a `.generated_files` file at the root of the project to utilize the apigentools feature of deleting known generated files between generation runs - https://apigentools.readthedocs.io/en/latest/special_files/

### Review checklist
Please check relevant items below:
- [ ] This PR includes all [newly recorded cassettes](/DEVELOPMENT.md) for any modified tests.

- [x] This PR does not rely on API client schema changes.
    - [ ] The CI should be fully passing.
- [ ] Or, this PR relies on API schema changes and this is a Draft PR that includes tests.
  - Note: CI shouldn't be run on this Draft PR, as its expected to fail without the corresponding schema changes.

### CODE REVIEW

File: .generated_files [hunk 1]
```diff
@@ -0,0 +1,2 @@
+api_docs/*
+src/main/java/com/datadog/api/*
```
Comment (jirikuncar) [inline .generated_files:2]: is it glob or regex?

Comment (nmuesch) [reply .generated_files]: Its regex - https://github.com/DataDog/apigentools/blob/master/apigentools/commands/generate.py#L426
Fixed :)

**Pull request merged**